dts/bindings/i2c/nxp,imx-i2c.yaml

```diff
@@ -0,0 +1,33 @@
+#
+# Copyright (c) 2018 Linaro Limited
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+---
+title: i.MX I2C Controller
+version: 0.1
+
+description: >
+    This is a representation of the i.MX I2C nodes
+
+inherits:
+    !include i2c.yaml
+
+properties:
+    compatible:
+      type: string
+      category: required
+      description: compatible strings
+      constraint: "nxp,imx-i2c"
```

Comment:
should be compatible = "fsl,imx7d-i2c";

Reply:
@galak,

This binding can be used by other imx cores (ie. imx6sx_m4). I believe that it is the same situation of nxp,imx-gpio.yaml, that shares the same dts binding and gpio driver shim.

I even think that they can share the same i2c driver shim. Maybe @stanislav-poboril can help us in checking this.

Reply:
So, do you think we have to change in nxp,imx-i2c.yaml to ```constraint: "fsl,imx7d-i2c", "fsl,imx21-i2c"``` and on nxp_imx7d_m4.dtsi ```compatible = "fsl,imx7d-i2c"```?

Sorry but I got a little bit confused what needs to be changed.

Reply:
If I change from "nxp,imx-i2c" to "fsl,imx21-i2c" I'll have to change NXP_IMX_I2C to FSL_IMX21_I2C in the dts.fixup file which will not follow the definition pattern of the other devices like UART and GPIO and looks strange to me.

Reply:
> So, do you think we have to change in nxp,imx-i2c.yaml to constraint: "fsl,imx7d-i2c", "fsl,imx21-i2c" and on nxp_imx7d_m4.dtsi compatible = "fsl,imx7d-i2c"?
> 
> Sorry but I got a little bit confused what needs to be changed.

So we don't handle multiple compats well today with regards to the binding files.  So I'm saying we use 'fsl,imx7d-i2c' so that we are matching linux (at least partially).

I understand that the pattern will differ from other items, but that is ok.  This is what should be in the DTS for the i2c controller.

Reply:
> So we don't handle multiple compats well today with regards to the binding files.

Does this mean we can define exactly one constraint in a given yaml file, but we can define multiple compatibles in the dts?

Reply:
> Does this mean we can define exactly one constraint in a given yaml file, but we can define multiple compatibles in the dts?

Correct, the dts is happy to have multiple compats.  The yaml for the binding don't handle this well.  Right now there is an expectation of 1:1 yaml to compat.

Reply:
@galak ,

Devicetree compatible string suggestion addressed.